__tests__/app.test.js

```diff
@@ -92,59 +92,117 @@ describe("GET /api/articles", () => {
-        const articelArray = JSON.parse(body)
-        expect(
-          articelArray.map((a) => (a.created_at = new Date(a.created_at).getTime()))
-        ).toBeSorted({ descending: true });
+        const articelArray = JSON.parse(body);
+        expect(articelArray.map((a) => a.created_at)).toBeSorted({
+          descending: true,
+        });
       });
   });
 });
 describe("GET /api/articles/:article_id/comments", () => {
   test("200:responds with an array of comments", () => {
     return request(app)
-      .get('/api/articles/1/comments')
+      .get("/api/articles/1/comments")
       .expect(200)
       .then((response) => {
         const body = response.text;
-        const commentArray = JSON.parse(body)
+        const commentArray = JSON.parse(body);
+        const expected = {
+          comment_id: expect.any(Number),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          article_id: expect.any(Number),
+        };
+        commentArray.forEach((comment) => {
+          expect(comment).toMatchObject(expected);
+        });
         expect(Array.isArray(commentArray)).toBe(true);
-      })
+      });
   });
   test("404:responds with an error, this article_id does not exist yet", () => {
     return request(app)
-      .get('/api/articles/100/comments')
+      .get("/api/articles/100/comments")
       .expect(404)
-        .then((response) => {
-          expect(response.body.msg).toBe("this article does not exist yet");
-        });
-    });
+      .then((response) => {
+        expect(response.body.msg).toBe("this article id does not exist yet");
+      });
+  });
+  test("200:valid input, with a recognised id, but reponds with empty array and appropiate message", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "this article does not have any comments"
+        );
+        expect(response.body.comments).toEqual([]);
+      });
+  });
 });
-test("404:responds with an error, this article_id does not have any comments", () => {
-  return request(app)
-    .get('/api/articles/2/comments')
-    .expect(404)
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201:responds with an object, a comment with properties of username and body", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "60% of the time it works every time",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
       .then((response) => {
-        expect(response.body.msg).toBe("this article does not have any comments");
+        expect(response.body).toEqual(
+          expect.objectContaining({
+            username: newComment.username,
+            body: newComment.body,
+          })
```

Comment:
This is a well laid out test 🥇 There is a debate with POST and PATCH about if the new object needs to send back everything in the new / updated object (see [here](https://softwareengineering.stackexchange.com/questions/314066/restful-api-should-i-be-returning-the-object-that-was-created-updated) for more info). The stance from Northcoders is we should send back all the information about the object. The majority seem in favour of this and recommend _at least_ sending the id of the new object back. ___I personally would recommend returning everything___!